### ISSUE TITLE
Query: SqlExpressionFactory doesn't apply type mapping to InExpression

### ISSUE THREAD

**Issue by thomaslevesque** [dotnet/efcore#26084]
I have an enum and an extension method like this:

```csharp
public enum State { None = 0, Foo = 1, Bar = 2, Baz = 3, ... };
public static bool IsValidState(this State state) => state == State.Foo || state == State.Bar;
```
I'd like to map this extension method to be able to use it in Linq queries. Basically, I'd like this :
```csharp
.Where(e => e.State.IsValidState())
```
to be translated to the following SQL :
```sql
WHERE [e].[State] IN (1, 2)
```

I tried this:

```csharp
            var validStates = new[] { State.Bar, State.Baz };
            modelBuilder.HasDbFunction(() => State.None.IsValidState())
                .HasTranslation(args => new InExpression(
                    args.First(),
                    new SqlConstantExpression(Expression.Constant(validStates), args.First().TypeMapping),
                    negated: false,
                    typeMapping: null));
```

But I still get an error saying the query can't be translated. I suspect it's because I didn't specify a type mapping, but I don't know what type mapping to provide.

How to do this?

EF Core version: 5.0.0
Database provider: Microsoft.EntityFrameworkCore.SqlServer
Target framework: .NET 5.0
Operating system: Windows
IDE: JetBrains Rider

**Comment by maumar** [comment]
workaround: 

```cs
.HasTranslation(args => new InExpression(
                    args.First(),
                    new SqlConstantExpression(Expression.Constant(validStates), args.First().TypeMapping),
                    negated: false,
                    typeMapping: new SqlServerBoolTypeMapping("bit", DbType.Boolean)));
```

**Comment by maumar** [comment]
@thomaslevesque yes, the code is supposed to work - we are supposed to fill the type mapping for you, and we do it for most of the expression types, but we missed InExpression. It should be a relatively simple fix and we expect to have it done before 6.0 ships.

**Comment by thomaslevesque** [comment]
Nice, thanks!
